### PR TITLE
created models, relations and validations

### PR DESCRIPTION
**Pull request by rofiuddin**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,47 +1,40 @@
...
+
+gem "rails_config"
+gem "devise"
+gem "rails",                "4.2.1"
+gem "faker",                "1.4.2"
+gem "carrierwave",             "0.10.0"
+gem "mini_magick",             "3.8.0"
+gem "fog",                     "1.36"
+
+gem "will_paginate",           "3.0.7"
```
Comment (MuhammadTamzid) [inline Gemfile]: use Kaminari instead of it

File: Gemfile [hunk 2]
```diff
@@ -1,47 +1,40 @@
...
+gem "jquery-rails",         "4.0.0.beta2"
+gem "turbolinks",           "2.3.0"
+gem "jbuilder",             "2.2.3"
+gem "sdoc",                 "0.4.0", group: :doc
 
 group :development, :test do
-  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
-  gem 'byebug'
+  gem "sqlite3",     "1.3.9"
+  gem "byebug",      "3.4.0"
```
Comment (MuhammadTamzid) [inline Gemfile]: use pry instead of it

File: app/models/admin.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class Admin < ActiveRecord::Base
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
```
Comment (MuhammadTamzid) [inline app/models/admin.rb]: indent

devise :database_authenticatable, :registerable,
  :recoverable, :rememberable, :trackable, :validatable

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class User < ActiveRecord::Base
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
```
Comment (MuhammadTamzid) [inline app/models/user.rb]: indent

File: db/schema.rb [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    t.integer  "project_id", limit: 4
+    t.integer  "user_id",    limit: 4
+    t.datetime "created_at",           null: false
+    t.datetime "updated_at",           null: false
+  end
+
+  create_table "projects", force: :cascade do |t|
+    t.string   "name",         limit: 255
+    t.string   "abbreviation", limit: 255
+    t.date     "startDate"
```
Comment (dieunb) [inline db/schema.rb]: `start_date`

File: db/schema.rb [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+    t.date     "endDate"
```
Comment (dieunb) [inline db/schema.rb]: `end_date`

File: Gemfile [hunk 7]
```diff
@@ -1,47 +1,39 @@
...
-# Use Unicorn as the app server
-# gem 'unicorn'
-
-# Use Capistrano for deployment
-# gem 'capistrano-rails', group: :development
+source "https://rubygems.org"
+
+gem "rails_config"
+gem "devise"
+gem "rails",                "4.2.1"
```
Comment (dieunb) [inline Gemfile]: latest rails version `4.2.5` pls

**Pull request merged**